DatadogInternal/Sources/Codable/AnyDecodable.swift

```diff
@@ -93,6 +93,8 @@ extension _AnyDecodable {
             self.init(double)
         } else if let string = try? container.decode(String.self) {
             self.init(string)
+        } else if let container = container as? PassthroughValueDecodingContainer, let value = try? container.decodePassthrough() {
+            self.init(value)
```

Comment:
introducing non-primitive types here I think we are trying to patch a problem that needs to be fixed somewhere else. 

Can we not pass `PassthroughValueDecodingContainer` between boundaries?

Reply:
IMO, 
```
 } else if let dictionary = try? container.decode([String: AnyDecodable].self) {
            self.init(dictionary.mapValues { $0.value })
        } 
```
should already cover this scenario. You can build any deep layer without any infinite loop.

Reply:
No it's not, `let dictionary = try? container.decode([String: AnyDecodable].self)` will call `AnyDecodable.init(from decoder: Decoder)` on each value and we need to be able to get a `PassthroughAnyCodable` value.

In addition to [these](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/Codable/AnyDecodable.swift#L76L94), the `AnyDecodable` needs to support `PassthroughAnyCodable` otherwise it will fail to decode [those](https://github.com/DataDog/dd-sdk-ios/blob/develop/DatadogInternal/Sources/Codable/AnyEncoder.swift#L675L678) and any additional passthrough values we may add.

Reply:
It's an optimisation to avoid any useless encoding on known types, essentially types that are shared in `DatadogInternal`, see https://github.com/search?q=repo%3ADataDog%2Fdd-sdk-ios+PassthroughAnyCodable&type=code

It's indeed more difficult to handle but it's best to minimize message-bus impact on perf, so I won't be removing it.

Reply:
> I honestly think, we made it difficult to handle by introducing "Passthrough" thing.

I share the same feeling. For me, the [number of abstractions](https://github.com/DataDog/dd-sdk-ios/tree/develop/DatadogInternal/Sources/Codable) in messaging and feature context coding becomes too high to understand this part well. The `PassthroughAnyCodable` seems to promise convenience, but it makes the compromise on clarity. I'd rather see it kept bare simple.